File: Node.js/getMyIp.js
// getMyIp.js - Fetch external IP and other info from curlmyip.ru
const axios = require('axios');

const BASE_URL = 'http://curlmyip.ru';

async function fetchEndpoints() {
    try {
        // Fetch IP address
        const { data: ip } = await axios.get(`${BASE_URL}/ip`);
        console.log(`Your IP address: ${ip}`);

        // Fetch User-Agent
        const { data: ua } = await axios.get(`${BASE_URL}/ua`);
        console.log(`Your User-Agent: ${ua}`);

        // Fetch and display headers
        const { data: headers } = await axios.get(`${BASE_URL}/headers`);
        console.log('Headers sent by you:');
        console.log(JSON.stringify(headers, null, 2));

    } catch (error) {
        console.error('Error during API call:', error.message);
    }
}

fetchEndpoints();

File: examples/Go/getmyip.go
// getmyip.go - Fetch external IP and headers from curlmyip.ru
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"time"
)

const baseURL = "http://curlmyip.ru"

func main() {
	client := http.Client{Timeout: 5 * time.Second}

	// Fetch IP address
	resp, err := client.Get(baseURL + "/ip")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error getting IP: %v\n", err)
		return
	}
	defer resp.Body.Close()
	body, _ := io.ReadAll(resp.Body)
	fmt.Printf("Your IP address: %s", body)

	// Fetch and pretty-print headers
	headersResp, err := client.Get(baseURL + "/headers")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error getting headers: %v\n", err)
		return
	}
	defer headersResp.Body.Close()
	headersBody, _ := io.ReadAll(headersResp.Body)

	var headers map[string]interface{}
	if err := json.Unmarshal(headersBody, &headers); err == nil {
		fmt.Println("Headers sent by you:")
		prettyJSON, _ := json.MarshalIndent(headers, "", "  ")
		fmt.Println(string(prettyJSON))
	} else {
		fmt.Printf("Headers raw response: %s\n", headersBody)
	}
}